### PR TITLE
refactor(basic-filter): Adapted for custom forms

### PR DESCRIPTION
**Pull request by rhuss**

### CODE REVIEW

File: src/app/store/integration-support.service.ts [hunk 1]
```diff
@@ -20,6 +20,7 @@ export class IntegrationSupportService {
   }
 
   getFilterOptions(integration: Integration): Observable<any> {
+    // Should be /{connectorId}/actions/{actionId}/filters/options
```
Comment (rhuss) [inline src/app/store/integration-support.service.ts]: that's a bogus comment

**Comment by rhuss** [comment]
@akieling @gashcrumb could you please have look ?

**Comment by rhuss** [comment]
sorry, please wait a bit. just messed up the PR.

**Comment by rhuss** [comment]
Ok, good to go for a review.

**Comment by gashcrumb** [comment]
@rhuss you've a lint error:

```
src/app/store/step/step.store.ts[198, 27]: expected nospace before colon in call-signature
```

you can run `yarn lint` locally.  Also it's possible to set up your IDE to fix this kind of stuff automatically on save, at least that's how I have vscode set up.  Then I just have to manually make sure I don't commit `console.log` statements all over the code :-)

File: src/app/store/step/step.store.ts [hunk 2]
```diff
@@ -192,4 +192,10 @@ export class StepStore {
   getSteps() {
     return this.steps;
   }
+
+  // Check if we need a custom form handling which stores the parsed
+  // properties in customProperties
+  isCustomStep(step: Step) : boolean {
+    return step.stepKind === BASIC_FILTER;
```
Comment (gashcrumb) [inline src/app/store/step/step.store.ts]: The data mapper should fall into this category too really.  If we set an empty `configuredProperties` on the step object in that case it shouldn't be the end of the world.  In which case we can reduce the code a bit more in the step-configure controller, WDYT?

Comment (rhuss) [inline src/app/store/step/step.store.ts]: If we could agree on a custom handling that would be awesome. However I found that the handling was a bit different, e.g. when the datamapper step finishes, a different navigation [happens](https://github.com/rhuss/syndesis-ui/blob/4d0760d8ad5e36845616bff93ed101a17cde1128/src/app/integrations/edit-page/step-configure/step-configure.component.ts#L68)

For https://github.com/rhuss/syndesis-ui/blob/4d0760d8ad5e36845616bff93ed101a17cde1128/src/app/integrations/edit-page/step-configure/step-configure.component.ts#L145 I think we can combine both easily.

**Pull request merged**